quick-start/main.go

```diff
@@ -22,37 +23,30 @@ func main() {
 	client.SetToken("dev-only-token")
 
 	secretData := map[string]interface{}{
-		"data": map[string]interface{}{
-			"password": "Hashi123",
-		},
+		"password": "Hashi123",
 	}
 
 	// Write a secret
-	_, err = client.Logical().Write("secret/data/my-secret-password", secretData)
+	_, err = client.KVv2("secret").Put(context.TODO(), "my-secret-password", secretData)
```

Comment:
Nit: It might be nice to use something like:
```go
	ctx, cancelContextFunc := context.WithCancel(ctx.Background())
	defer cancelContextFunc()
	...
	_, err = client.KVv2("secret").Put(ctx, "my-secret-password", secretData)
```

Reply:
WithCancel might not be necessary here. WithTimeout might be useful though if we want to limit these to run within ~10seconds or so.

Reply:
I can put these in this version of the quickstart since this repo represents a more "built-out" version of our other examples in general, but for the docs snippets and actual quickstart guide page I think I'll just stick to plain `context.Background()` to keep it simple and not too overwhelming for those who might be less familiar with the ins and outs of the `context` package.

Reply:
Yup! My main consideration was that this is basically an app, as opposed to a collection of snippets, so `context.TODO()` didn't seem like a good fit. Maybe even something simple just to improve readability:
```go
        ctx := context.Background()
	...
	_, err = client.KVv2("secret").Put(ctx, "my-secret-password", secretData)
	_, err = client.KVv2("secret").Get(ctx, "my-secret-password")
```